src/Controls/src/SourceGen/RouteSyntaxReceiver.cs

```diff
@@ -0,0 +1,267 @@
+								{
+									case "Routes" when argument.Expression is ArrayCreationExpressionSyntax arrayExpression:
+										if (arrayExpression?.Initializer?.Kind() == SyntaxKind.ArrayInitializerExpression)
+										{
+											routes.AddRange(arrayExpression.Initializer.Expressions.Select(expr => GetRoute(cds, expr)));
+										}
+										break;
+									case "Routes" when argument.Expression is ImplicitArrayCreationExpressionSyntax implicitArrayExpression:
+										if (implicitArrayExpression.Initializer.Kind() == SyntaxKind.ArrayInitializerExpression)
+										{
+											routes.AddRange(implicitArrayExpression.Initializer.Expressions.Select(expr => GetRoute(cds, expr)));
+										}
+										break;
+									case "Lifetime":
+										lifetime = GetLifetime(argument.Expression);
+										break;
+									case "ImplicitViewModel" when argument.Expression is LiteralExpressionSyntax literalExpression:
+										_ = bool.TryParse(literalExpression.Token.ValueText, out implicitViewModel);
+										break;
+									case "ViewModelType" when argument.Expression is TypeOfExpressionSyntax typeOfExpression:
+										viewModelType = ((IdentifierNameSyntax)typeOfExpression.Type).Identifier.Text;
+										break;
+								}
+							}
+							else if (argument.Expression is not null)
+							{
+								if (routeArgument)
+								{
+									routeArgument = false;
+									route = GetRoute(cds, argument.Expression);
+								}
+								else
+								{
+									lifetime = GetLifetime(argument.Expression);
+								}
+							}
+						}
+					}
+
+					if (routes.Count == 0)
+					{
+						routes.Add(route);
+					}
+
+					_routedPages.Add(new RoutedPage(cds, routes, lifetime, implicitViewModel, viewModelType));
+				}
+
+				if (serviceAttribute is not null)
+				{
+					arguments = serviceAttribute.ArgumentList?.Arguments;
+
+					lifetime = Singleton;
+					var registerFor = string.Empty;
+					var useTryAdd = false;
+
+					if (arguments is not null)
+					{
+						foreach (var argument in arguments)
+						{
+							if (argument.NameColon is not null)
+							{
+								if (argument.NameColon.Name.Identifier.Text == "lifetime")
+								{
+									lifetime = GetLifetime(argument.Expression);
+								}
+							}
+							else if (argument.NameEquals is not null)
+							{
+								switch (argument.NameEquals.Name.Identifier.Text)
+								{
+									case "RegisterFor" when argument.Expression is TypeOfExpressionSyntax typeOfExpression:
+										registerFor = ((IdentifierNameSyntax)typeOfExpression.Type).Identifier.Text;
+										break;
+									case "UseTryAdd" when argument.Expression is LiteralExpressionSyntax literalExpression:
+										_ = bool.TryParse(literalExpression.Token.ValueText, out useTryAdd);
+										break;
+								}
+							}
+							else if (argument.Expression is not null)
+							{
+								lifetime = GetLifetime(argument.Expression);
+							}
```

Comment:
This is the concatenation of tree if with no final else clause, so we can just remove all the else clauses used here. can be 3 ifs one after the other